js/content/community.js

```diff
@@ -3670,42 +3671,75 @@ let WorkshopBrowseClass = (function(){
                         window.dialog.Dismiss();
                     }
 
-                    window.dialog = ShowBlockingWaitDialog("${Localization.str[method + "_all"]}", \`${Localization.str[method + "_loading"].replace("__i__", ++i).replace("__count__", total)}\`);
+                    window.dialog = ShowBlockingWaitDialog("${Localization.str[method + "_all"]}", \`${Localization.str[method + "_loading"].replace("__i__", completed).replace("__count__", total)}${failed ? ` (${Localization.str.failed.replace("__n__", failed)})` : ""}\`);
                 }`)
             }
 
             function changeSubscription(id) {
-                return new Promise(function(resolve) {
-                    let formData = new FormData();
-                    formData.append("sessionid", User.getSessionId());
-                    formData.append("appid", appid);
-                    formData.append("id", id);
-
-                    RequestData.post("https://steamcommunity.com/sharedfiles/" + method, formData, {
-                        withCredentials: true
-                    }).then(function() {
-                        updateWaitDialog();
-                        resolve();
-                    });
-                });
+                let formData = new FormData();
+                formData.append("sessionid", User.getSessionId());
+                formData.append("appid", appid);
+                formData.append("id", id);
+
+                return RequestData.post("https://steamcommunity.com/sharedfiles/" + method, formData, {
+                    withCredentials: true
+                }, true)
+                .then(function(res) {
+                    if (!res || !res.success || res.success !== 1) {
+                        failed++;
+                        console.error(new Error("Bad response"))
```

Comment:
You can `throw` here, the `catch` callback will handle this

Reply:
Resolved in https://github.com/tfedor/AugmentedSteam/pull/506/commits/0c3fd463f0c96a4ad74c16e14489c4e2f6baa767